addons/fluentbit/1.5.x/fluentbit-4.yaml

```diff
@@ -35,10 +35,13 @@ spec:
     version: 0.6.1
     values: |
       service:
-        annotations:
-          prometheus.io/path: "/api/v1/metrics/prometheus"
-          prometheus.io/port: "2020"
-          prometheus.io/scrape: "true"
+        annotations: {}
+      serviceMonitor:
```

Comment:
Could you give me a quick explanation for this change? Have you tested this out with prometheus?

Reply:
Sure.
So first added annotations where for pod monitoring / scraping. These didn't worked as they we're not complete for our Prometheus operator setup.

As the official fluent-bit chart has a good Service Monitor enablement logic (with checking for api capability) I checked out we can really use it in this way. And it works for our Prometheus Operator addon.
So we we're able to drop the annotations as not used and use ServiceMonitor.

Reply:
thank you for that explanation.

Reply:
Just FYI, this would have had to have been a service label `servicemonitor.kubeaddons.mesosphere.io/path: "api__v1__metrics__prometheus"` and maybe a change to prom config to add the port for it to work, but it's better to use the servicemonitor defined in fluentbit since it actually checks [`.Capabilities.APIVersions.Has "monitoring.coreos.com/v1"`](url)

Reply:
> ... it actually checks .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" ...

It does? Oh no. That means that it won't install the service monitor if prometheus isn't installed first.

Reply:
Lol right, back to this recurring chart-dependency issue 🙈 Do we want to just be consistent from now on about defining **all** servicemonitors within prometheus-operator until we come up with a better solution? There are a couple addons that don't - velero and istio are two that I can think of

Reply:
Unfortunately, [fluent-bit currently doesn't support adding service labels](https://github.com/fluent/helm-charts/blob/master/charts/fluent-bit/templates/service.yaml#L5-L6) so we'd have to open up an upstream pr to add that first. In my experience (I've had to do this a couple times for upstream charts for precisely this reason - add labels for prom service monitor :p) the change is accepted without any issue. Depending on how active the chart maintainers are, it can be a quick merge (or not, if it's hard to get ahold of them)

Reply:
We'll see 🙃 https://github.com/fluent/helm-charts/pull/27

Reply:
^ Merged! I'll bump the chart to `0.6.2`

Reply:
we'll also need to add an endpoint to the service monitor def in prometheus-operator [here](https://github.com/mesosphere/kubernetes-base-addons/blob/master/addons/prometheus/0.38.x/prometheus-12.yaml#L109) something like
```
              - path: /api/v1/metrics/prometheus
                port: http
                interval: 30s
```
to get metrics scraped

Reply:
> It does? Oh no. That means that it won't install the service monitor if prometheus isn't installed first.

@joejulian 
What is the point?
Fluent-bit comes long after elasticsearch And that for after prometheus. We can make fluent-bit also dependent to have prometheus installed  before. We should not move ServiceMonitors out of responsible charts.


> we'll also need to add an endpoint to the service monitor def in prometheus-operator here something like
That is **not** true @gracedo

Reply:
@gracedo why isn't something like
``` 
servicemonitor.kubeaddons.mesosphere.io/path: "api__v1__metrics__prometheus"
servicemonitor.kubeaddons.mesosphere.io/port: "http"
```

without any extra change to another chart or addon?

Reply:
@gracedo added more labels to define it clearer also in `prometheus`(-operator) addon